src/clarity-angular/select/interfaces/state.ts

```diff
@@ -0,0 +1,6 @@
+/*
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+export interface State { search: string; }
```

Comment:
Well on second thought or actually because i've now changed it. A problem with using `ngModel` here is that on a selection `ngModel` becomes an instance of `Option` instead of a `String` that rises a few issues.